src/ray/raylet/node_manager.cc

```diff
@@ -2319,7 +2320,7 @@ bool NodeManager::FinishAssignedTask(Worker &worker) {
   worker.ResetTaskResourceIds();
 
   const auto &spec = task.GetTaskSpecification();
-  if (spec.IsActorCreationTask() || spec.IsActorTask()) {
+  if ((spec.IsActorCreationTask() || spec.IsActorTask()) && !spec.IsDirectCall())  {
```

Comment:
actually direct call tasks won't be assigned by raylet, so this additional check is not necessary?

Reply:
It still goes through here when returning the worker lease.

Reply:
This reminds me of another thing:  we need to make sure that we return the lease to raylet when the task finished is an actor creation task, otherwise the caller might still push general tasks to it which will be incorrect. Not sure if this is already handled or not.

Reply:
Just double check and it's indeed not handled yet. I've just fixed this by returning the worker to raylet when it just finishes an actor creation task.

Reply:
@ericl  is this added check necessary? Would the existing logic in `FinishAssignedActorTask` work for direct actor creation?